src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -127,8 +155,21 @@ public Void run() {
 
             if (v == null) {
                 int keepAliveTimeout = http.getKeepAliveTimeout();
-                v = new ClientVector(keepAliveTimeout > 0 ?
-                        keepAliveTimeout * 1000 : LIFETIME);
+                if (keepAliveTimeout == 0) {
+                    keepAliveTimeout = getUserKeepAlive(http.getUsingProxy());
```

Comment:
This could be 0 if -Dhttp.keepAlive.time.xxx=0 was specified which would result in an assert below
I am not sure I understand the logic of having same 5s timeout for server & proxy if nothing was specified anywhere, but having a different value for proxy & server if the server specified keep-alive without providing a value.
Where does that come from?

Reply:
Ah, I meant to catch that case as well in the method where the property is read. Probably should be documented as well.

The other behavior is just maintaining the present behavior, purely for compatibility reasons.